cmake/extensions.cmake

```diff
@@ -689,6 +689,80 @@ function(zephyr_check_compiler_flag lang option check)
   endif()
 endfunction()
 
+
+# add_named_section(section_name region)
+#
+# This function creates a linker section with the given name placed in the given
+# region, and creates symbols for the start, end, and size of the section.
+#
+# See also add_named_ro_section() and add_named_rw_section() to avoid specifying
+# the region.
+#
+# This function will give an error if the same section name is added multiple
+# times, since code will often expect only its own data to be present in the
+# section.
+#
+# Example 1:
+# add_named_section(".foo", RAMABLE_REGION)
+# creates a section named .foo in RAM, and the symbols _foo_start, _foo_end, and
+# _foo_size. Notice that the '.' is dropped from the symbol names.
+#
+# To access the symbols from code:
+#   extern const mytype_t _foo_start;
+#   extern const u32_t _foo_size;
+#   ...
+#   do_something_with_foo(&_foo_start, (u32_t)&_foo_size);
+#
+# To add c variables to the section:
+#   const mytype_t var_placed_in_foo _GENERIC_SECTION(.foo) __attribute__((used)) = something;
+#
+# Example 2:
+# add_named_section("bar", ROMABLE_REGION)
+# creates a section named bar in flash, and the symbols _bar_start, _bar_end,
+# and _bar_size.
+#
+set(generated_sections_filename "${PROJECT_BINARY_DIR}/include/generated/generated-sections.ld")
+file(WRITE ${generated_sections_filename} "")
```

Comment:
So far, unless I've missed a patch, we have been able to limit this file to functions and macros. I'd like this to continue to hold if possible.

Reply:
Done